src/libraries/System.Private.CoreLib/src/System/Globalization/DateTimeFormat.cs

```diff
@@ -723,11 +732,10 @@ private static bool IsUseGenitiveForm(ReadOnlySpan<char> format, int index, int
                         break;
 
                     default:
-                        // NOTENOTE : we can remove this rule if we enforce the enforced quote
-                        // character rule.
+                        // NOTENOTE : we can remove this rule if we enforce the enforced quote character rule.
                         // That is, if we ask everyone to use single quote or double quote to insert characters,
                         // then we can remove this default block.
-                        result.Append(TChar.CastFrom(ch));
+                        AppendChar(ref result, ch);
```

Comment:
@tarekgh @stephentoub if we'd want to backport the non-breaking space support in DateTime. In addition to https://github.com/dotnet/runtime/pull/83589, we only need this line, right?

Reply:
Backport to what? This line was only introduced last week.

Reply:
libicu changed to use a different space in the short time format for en_US.
This space wasn't handled previously, so it disappears in distros with a recent icu (like Fedora 38).
I fixed that with https://github.com/dotnet/runtime/issues/83571.

> This line was only introduced last week.

I assumed the failures on Fedora 38 for these tests happened due to my change in https://github.com/dotnet/runtime/issues/83571 which lets the unicode space to propagate instead of replacing it.

I didn't know this cast was only recently introduced.